tests/s25Main/campaign/testCampaignSaveData.cpp

```diff
@@ -0,0 +1,102 @@
+// Copyright (C) 2024 Settlers Freaks (sf-team at siedler25.org)
+//
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "CampaignSaveData.h"
+#include "Settings.h"
+#include "lua/CampaignDataLoader.h"
+#include "gameData/CampaignDescription.h"
+#include "gameData/CampaignSaveCodes.h"
+#include "rttr/test/TmpFolder.hpp"
+#include <boost/nowide/fstream.hpp>
+#include <boost/test/unit_test.hpp>
+
+namespace {
+constexpr auto uid = "uniqueID";
+struct CampaignSaveDataFixture
+{
+    CampaignSaveDataFixture()
+    {
+        desc.uid = uid;
+        saveData.clear();
+    }
+
+    CampaignDescription desc;
+    std::map<std::string, std::string>& saveData = SETTINGS.campaigns.saveData;
+};
+} // namespace
+
+BOOST_FIXTURE_TEST_CASE(isChapterEnabled_ReturnsFalse_WhenChapterIsDisabled, CampaignSaveDataFixture)
+{
+    BOOST_TEST_REQUIRE(isChapterEnabled(desc, 0) == false);
+    BOOST_TEST_REQUIRE(isChapterEnabled(desc, 1) == false);
+    BOOST_TEST_REQUIRE(isChapterEnabled(desc, 2) == false);
+    BOOST_TEST_REQUIRE(isChapterEnabled(desc, 9) == false);
+}
+
+BOOST_FIXTURE_TEST_CASE(isChapterEnabled_ReturnsTrue_WhenChapterIsEnabled, CampaignSaveDataFixture)
+{
+    saveData[uid] = "011";
+    BOOST_TEST_REQUIRE(isChapterEnabled(desc, 0) == false);
+    BOOST_TEST_REQUIRE(isChapterEnabled(desc, 1) == true);
+    BOOST_TEST_REQUIRE(isChapterEnabled(desc, 2) == true);
+    BOOST_TEST_REQUIRE(isChapterEnabled(desc, 9) == false);
+}
+
+BOOST_FIXTURE_TEST_CASE(Chapters0and1AreEnabledByDefault, CampaignSaveDataFixture)
+{
+    saveData[uid] = CampaignSaveCodes::defaultChaptersEnabled;
```

Comment:
So the default is to enable the first 2 chapters? Why that? Wouldn't (only) the first one be a more sensible default?
I'm also a bit confused as you seem to be using indices **and** 1-based values for chapters

Maybe stick to indices and use an optional instead of zero